### ISSUE TITLE
Option to make the nets more deterministic

### ISSUE THREAD

**Issue by run2** [dnouri/nolearn#26]
What do you think about introducing an option to seed the random before doing the KFold on train test split ? That way the net predictions and loss details will be more deterministic over multiple runs on the same set.
Thanks

**Comment by dnouri** [comment]
I think right now you can do a `np.random.seed(42)` before training and it'll be deterministic?

**Comment by run2** [comment]
Sure yes. I just wanted to check if you want to add it as a option. I will close this and you can reopen it if and when you want to.

**Comment by cancan101** [comment]
The default for `KFold` is actually NOT to shuffle (i.e. it is deterministic):
https://github.com/scikit-learn/scikit-learn/blob/38104ff4e8f1c9c39a6f272eff7818b75a27da46/sklearn/cross_validation.py#L320

**Comment by dnouri** [comment]
Ah yes, very useful if your data is not [independently distributed](http://scikit-learn.org/stable/modules/cross_validation.html#a-note-on-shuffling).

**Comment by cancan101** [comment]
It might be worth tossing a note along the lines of:
>By default no shuffling occurs, including for the (stratified) K fold cross- validation....

**Comment by dnouri** [comment]
Yes, you're right.  And that reminds me that I should be working on proper docstrings soon.

**Comment by neilsummers** [comment]
From what I can see there is more randomness than just in the train test splits. From the tests I have done there is randomness when I include a dropout layer. I can't seem to make this the same for each run by doing a np.random.seed(42) before the run. I have tried tracing it back throug the source, and it appears to be setting a seed by default in RandomStreams from theano, which is called in the DropoutLayer, but I am still getting changes from run to run when I include a DropoutLayer, but no changes when there is no DropoutLayer.

**Comment by dnouri** [comment]
There's an issue for that in Lasagne: https://github.com/Lasagne/Lasagne/issues/6
